pagination-practice/src/components/PaginatedTable.jsx

```diff
@@ -0,0 +1,182 @@
+import { useEffect, useState } from "react";
+import axios from "axios";
+export default function PaginatedTable() {
+  const [data, setData] = useState([]);
+  const [currentData, setCurrentData] = useState([]);
+  const [currentPageIndex, setCurrentPageIndex] = useState(0);
+  const [pageSize, setPageSize] = useState(10);
+
+  const pageCount = Math.ceil(data.length / pageSize);
+
+  const indexOfFirstItem = pageSize * currentPageIndex;
+  const indexOfLastItem = Math.min(
+    indexOfFirstItem + pageSize,
+    data.length - 1
+  );
+
+  function gotoPage(pageIndex) {
+    if (pageIndex >= 0 && pageIndex < pageCount) {
+      setCurrentPageIndex(pageIndex);
+    }
+  }
+
+  function canGotoPreviousPage() {
+    if (currentPageIndex > 0) {
```

Comment:
can shorten this "return currentPageIndex > 0;"

Reply:
updated.